### PR TITLE
GNA SF propagation fix

### PR DESCRIPTION
**Pull request by sirzabek**
### Details:
 - disable weights scale factor change for crop layer
 - fix handling of 3D activations.

### Tickets:
 - 86028

### CODE REVIEW

File: src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp [hunk 1]
```diff
@@ -11,12 +11,32 @@ namespace LayerTestsDefinitions {
...
     void SetUp() override {
-        ActivationLayerTest::SetUp();
+        InferenceEngine::Precision netPrecision;
+        std::pair<std::vector<size_t>, std::vector<size_t>> shapes;
+        std::pair<ngraph::helpers::ActivationTypes, std::vector<float>> activationDecl;
+        std::tie(activationDecl, netPrecision, inPrc, outPrc, inLayout, outLayout, shapes, targetDevice) = GetParam();
+
+        auto inputShape = shapes.first;
+        const std::size_t inputDim = std::accumulate(inputShape.begin(), inputShape.end(), 1, std::multiplies<size_t>());
+        std::vector<size_t> inputDims{1, inputDim};
```
Comment (t-jankowski) [inline src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp:21]:
```suggestion
        const std::vector<size_t> inputDims{1, inputDim};
```

File: src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp [hunk 2]
```diff
@@ -11,12 +11,32 @@ namespace LayerTestsDefinitions {
...
+        activationType = activationDecl.first;
+        auto constantsValue = activationDecl.second;
```
Comment (t-jankowski) [inline src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp:23]:
```suggestion
        const auto& constantsValue = activationDecl.second;
```

File: src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp [hunk 3]
```diff
@@ -11,12 +11,32 @@ namespace LayerTestsDefinitions {
...
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
```
Comment (t-jankowski) [inline src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp:24]:
```suggestion
        const auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
```

File: src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp [hunk 4]
```diff
@@ -11,12 +11,32 @@ namespace LayerTestsDefinitions {
...
-        if (ngPrc == "u8" || ngPrc == "i16") {
+        if (ngPrc == ngraph::element::u8 || ngPrc == ngraph::element::i16) {
             threshold = 1.0;
         }
+
+        auto inputReshapePattern = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{inputShape.size()}, inputShape);
+        auto inputReshape = std::make_shared<ngraph::opset1::Reshape>(params[0], inputReshapePattern, false);
+        auto activation = ngraph::builder::makeActivation(inputReshape, ngPrc, activationType, shapes.second, constantsValue);
+        auto outputReshapePattern = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{2}, inputDims);
+        auto outputReshape = std::make_shared<ngraph::opset1::Reshape>(activation, outputReshapePattern, false);
```
Comment (t-jankowski) [inline src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp:37]: Use `const` whenever possible.

Comment (sirzabek) [reply src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp:37]: We should rewrite half of the code :-)

Comment (t-jankowski) [reply src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp:37]: My aim is to put attention on this aspect in new or being changed code. I'm not asking to rewrite PR unrelated pieces of code.

Comment (sirzabek) [reply src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp]: Sure. Mine was just a comment that actually we _should_ rewrite the unrelated pieces of code. Anyway, all requested changes done.

**Review by t-jankowski — APPROVED** [review]

**Review by kbruniec — APPROVED** [review]
LGTM

**Pull request merged**